notion-server/src/api/mod.rs

```diff
@@ -78,3 +80,29 @@ async fn create_execution_environment(
 
     Ok(status::Created(location.to_string(), None))
 }
+
+#[tokio::main]
+#[post("/shards", data = "<shard>")]
+async fn create_shard_chain(
+    shard: Json<args::CreateShardChain>,
+    handle: State<Handle>,
+) -> EthResult<status::Created<()>> {
+    let shard = shard.into_inner();
```

Comment:
I assume this just converts the `Json` object into the internal `CreateShardChain` arg?

Reply:
Yep! The `Json` struct wraps the `CreateShardChain` struct to give Rocket a hint about how to deserialize it. There are other wrappers for other encodings.